fineract-provider/src/main/java/org/apache/fineract/infrastructure/bulkimport/importhandler/ImportHandlerUtils.java

```diff
@@ -53,8 +53,14 @@ public static Integer getNumberOfRows(Sheet sheet, int primaryColumn) {
         Integer noOfEntries = 0;
         // getLastRowNum and getPhysicalNumberOfRows showing false values
         // sometimes
-        while (sheet.getRow(noOfEntries + 1) != null && sheet.getRow(noOfEntries + 1).getCell(primaryColumn) != null) {
-            noOfEntries++;
+        while (sheet.getRow(noOfEntries + 1) != null) {
```

Comment:
I would recommend a slightly different approach for better readability and performance:
```
while (true) {
    Row row = sheet.getRow(noOfEntries + 1);
    if (row == null) break;

    Cell cell = row.getCell(primaryColumn);
    if (cell == null || cell.getCellType() == CellType.BLANK) break;

    noOfEntries++;
}
```

- Storing in a variable the value avoids fetching the data multiple times
- 2 exit points to avoid unnecessary steps

Kindly let me know your thoughts!